scripts/astrokat-coords.py

```diff
@@ -0,0 +1,194 @@
+                                 float_format,
+                                 degree_format)
+
+
+def display_radec(astropy_radec, name=None):
+    if name is not None:
+        print('Target {}'.format(name))
+    print('(RA, DEC) coordinates')
+    print('\t{}'.format(coord_str(astropy_radec.ra,
+                                  astropy_radec.dec)))
+
+
+def display_galactic(astropy_gal, name=None):
+    if name is not None:
+        print('Target {}'.format(name))
+    print('Galactic coordinates')
+    print('\t{}'.format(coord_str(astropy_gal.l,
+                                  astropy_gal.b)))
+
+
+def csv_entry(radec_coord, name='target'):
+    x_str, y_str = format_str(radec_coord.ra, radec_coord.dec)
+    return '{}, radec target, {}, {}\n'.format(name,
+                                               x_str,
+                                               y_str)
+
+
+def gal2radec(infile, verbose=False):
+    """Read all targets in catalogue and make all ICRS in returned catalogue"""
+    with open(infile, 'r') as fin:
+        targets = fin.readlines()
+
+    if verbose:
+        print('\nInput file')
+        for target in targets:
+            print(target.strip())
+
+    cat_entry = ''
+    for target in targets:
+        if target[0] == '#':  # comment
+            continue
+        [name, tag, coord_x, coord_y] = [item.strip()
+                                         for item in target.strip().split(',')]
+        if 'radec' in tag:
+            radec_coord = icrs_skycoord(coord_x, coord_y)
+        else:
+            [gal_coord,
+             radec_coord] = gal2icrs_skycoord(coord_x, coord_y)
+
+        cat_entry += csv_entry(radec_coord, name=name)
+
+    return cat_entry
+
+
+def main(args):
+    if args.radec is not None:
+        radec_coord = icrs_skycoord(args.radec[1], args.radec[2])
+        display_radec(radec_coord, name=args.radec[0])
+        # exit here
+        raise SystemExit
+    elif args.galactic is not None:
+        [gal_coord,
+         radec_coord] = gal2icrs_skycoord(args.galactic[1],
+                                          args.galactic[2])
+        display_galactic(gal_coord, name=args.galactic[0])
+        display_radec(radec_coord)
+        cat_entry = csv_entry(radec_coord, name=args.galactic[0])
+    elif args.catalogue is not None:
+        cat_entry = gal2radec(args.catalogue,
+                              verbose=args.verbose)
+    else:
+        # nothing to do
+        raise SystemExit
+
+    if args.outfile is not None:
+        with open(args.outfile, 'w') as fout:
+            fout.write(cat_entry)
+        print('\nEquatorial catalogue file created: {}'.format(args.outfile))
+
+    if args.verbose:
+        print('Catalogue file entries')
+        print(cat_entry)
+
+
+if __name__ == '__main__':
+    # This code will take care of negative values for declination
+    for i, arg in enumerate(sys.argv):
+        if (arg[0] == '-') and arg[1].isdigit():
```

Comment:
May also raise an exception if len(arg) < 2.

Reply:
Very good suggestion, much appreciated
I updated all the scripts that use this little hack to add this extra test